draft-ietf-oauth-browser-based-apps.md

```diff
@@ -203,6 +203,119 @@ For this reason, and from other lessons learned, the current best practice for b
+- Execute malicious JS code
+- Send a request to a resource server and process the response
+
+To authorize the requests to the resource server, the attacker simply mimics the behavior of the client application. For example, when a client application programmatically attaches an access token to outgoing requests, the attacker does the same. Should the client application rely on an external component to augment the request with the proper access token, then this external component will also augment the attacker's request.
+
+This attack pattern is well-known and also occurs with traditional applications using HttpOnly session cookies. It is commonly accepted that this scenario cannot be stopped or prevented by application-level security measures. For example, the DPoP specification ({{dpop}}) explicitly considers this attack scenario to be out of scope.
+
+
+
+Attack Consequences
+-------------------
+
+Successful execution of a malicious payload can result in the theft of access tokens and refresh tokens, or in the ability to hijack the client application running in the user's browser. Each of these consequences is relevant for browser-based OAuth clients. They are discussed below in decreasing order of severity.
+
+
+### Exploiting Stolen Refresh Tokens
+
+When the attacker obtains a valid refresh token from a browser-based OAuth client, they can abuse the refresh token by running a Refresh Token flow with the authorization server. The response of the Refresh Token flow contains an access token, which gives the attacker the ability to access protected resources (See {{consequence-at}}). In essence, abusing a stolen refresh token enables long-term impersonation of the user to resource servers.
+
+The attack is only stopped when the authorization server refuses a refresh token for expiration reasons, or when the refresh token is revoked. In a typical browser-based OAuth client, it is not uncommon for a refresh token to remain valid for multiple hours, or even days.
+
+
+
+### Exploiting Stolen Access Tokens {#consequence-at}
+
+If the attacker obtains a valid access token, they gain the ability to impersonate the user in a request to a resource server. Concretely, possession of an access token allows the attacker to send arbitrary requests to any resource server that considers the access token to be valid. In essence, abusing a stolen access token enables short-term impersonation of the user to resource servers.
+
+The attack ends when the access token expires, or, in case of a reference token, when a token is revoked with the authorization server. In a typical browser-based OAuth client, access token lifetimes can be quite short, ranging from minutes to hours.
+
+Note that the possession of the access token gives allows its unrestricted use by the attacker. The attacker can send arbitrary requests to resource servers, using any HTTP method, destination URL, header values, or body.
+
```

Comment:
This should include a paragraph discussing DPoP, to mention both:

* DPoP does effectively prevent the attacker from using a stolen token (if bound to a nonexportable key), but
* the attacker may be able to obtain their own DPoP-bound tokens as described in {{payload-new-flow}}